### PR TITLE
GDAL_VERSION configures gdalversion

### LINKED ISSUE #364: Print an error msg if neither gdal-config or `--gdalversion` indicate a GDAL API version
Right now, `setup.py` raises a `NameError`:

```
Traceback (most recent call last):
   File "<string>", line 1, in <module>
   File "/tmp/pip-build-wfl5rbpb/fiona/setup.py", line 193, in <module>
     if gdalversion.startswith("1"):
  NameError: name 'gdalversion' is not defined
```

### PR DESCRIPTION
**Pull request by sgillies**
Defaults to "2".

Closes #364

### CODE REVIEW

**Comment by geowurster** [comment]
@sgillies Will review over the weekend.

**Comment by sgillies** [comment]
@geowurster can you review this one for me?

The intent: exit the setup script with a fatal error message if we have no GDAL API version. A value of "2" is the default, so that should rarely happen. Configuration via a GDAL_VERSION environment variable has been added.

File: setup.py [hunk 1]
```diff
@@ -121,14 +119,37 @@ def run(self):
...
                      "to locate needed GDAL files.\nMore information is "
                      "available in the README.")
         else:
-            log.warning("Failed to get options via gdal-config: %s", str(e))
+            log.warn("Failed to get options via gdal-config: %s", str(e))
+
+    # Get GDAL API version from environment variable.
+    gdalversion = os.environ.get('GDAL_VERSION', gdalversion)
+    if gdalversion:
+        log.info("GDAL API version obtained from command line option: %s",
```
Comment (geowurster) [inline setup.py]: @sgillies Typo in the log message - should state the version was obtained from the env variable.

**Comment by geowurster** [comment]
@sgillies Looks good aside from the log typo.

**Comment by sgillies** [comment]
Done, @geowurster (pending checks above).

**Pull request merged**